### PR TITLE
Add apis to fetch given pokemon stat and display

### PR DESCRIPTION
**Pull request by TaoXieSZ**
Hi there, I am a new player of pokemon showdown and found your wonderful plugin days ago! But I found it is also useful to display stats of each pokemon for new players.

### CODE REVIEW

**Comment by TaoXieSZ** [comment]
Example:
image

However, the pokemon api fails to fetch some pokemons like Basculegion or Darmanitan. Maybe we need more tests

File: src/pokeAPI.ts [hunk 1]
```diff
@@ -11,12 +11,39 @@ export namespace PokeAPI {
...
       if (this.cache[url] === undefined) {
-        const response = await axios.get(url);
-        this.cache[url] = response.data;
+        try{
+          const response = await axios.get(url);
+          this.cache[url] = response.data;
+        } catch (e) {
+          console.error('failed to fetch', url, e)
+          return null;
+        }
```
Comment (coltonb) [inline src/pokeAPI.ts:46]: Catching the error and returning `null` here is really broad. Returning `null` in the case of 404 might be OK, but I'd rather let errors bubble up and be caught on the highest level.

File: src/pokeAPI.ts [hunk 2]
```diff
@@ -11,12 +11,39 @@ export namespace PokeAPI {
...
+          name: data.name,
+          stats: data.stats.map((stat: any) => ({
+            name: stat.stat.name,
+            base_stat: stat.base_stat,
+            effort: stat.effort,
+          })),
+        };
+      } else {
+        console.log('no data found for', pokemon);
+        throw new Error(`No data found for ${pokemon}`);
```
Comment (coltonb) [inline src/pokeAPI.ts:32]: Catching the error and returning `null` is redundant in the caching `get` method if we're just raising a new error here when the response is `null` - if I understand correctly, the code functions identically without these changes, and the final error raised is missing the context that axios provides.

If the goal here was to more cleanly log what is going on for debug purposes, I might recommend updating the get method as such:

```typescript
  private async get(url: string) {
    if (this.cache[url] !== undefined) {
      let response;

      try {
        response = await axios.get(url);
      } catch (error) {
        console.error(`[${error.response.status}] PokeAPI request failed: GET ${url}`)
        throw error;
      }

      this.cache[url] = response.data;
    }

    return this.cache[url];
  }
```

**Review by coltonb — CHANGES_REQUESTED** [review]
I think this is a good idea, and I appreciate the clean implementation!

Do you think you could format the stats so that they are inline and smaller so as to make the change a little less intrusive?

I am thinking something like 

```
HP: 999    Atk: 999    Def: 999    SpA: 999    SpD: 999    Spe: 999
x4: ELECTR
...
```

Without a horizontal rule?

File: src/pokeAPI.ts (hunk 2 above)
Comment (TaoXieSZ) [reply src/pokeAPI.ts:32]: `error.response` is invalid. So I just attach the whole error in the log.

**Comment by TaoXieSZ** [comment]
Thanks for your comment! Because this is my first time to develop chrome extension, your comments are very useful to me.

I reorganize the tooltip as this:
image

**Review by coltonb — APPROVED** [review]
Looks great! Merging 👍 

Will try to get this pushed to the web store today, if not, expect it to be live sometime next week!

**Comment by TaoXieSZ** [comment]
Happy to contribute! Hope we can reach K8 ASAP!

**Pull request merged**